File: DarkOrbit 10.0/Net/netty/commands/DestructionTypeModule.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Ow.Utils;

namespace Ow.Net.netty.commands
{
    class DestructionTypeModule
    {
        public const short ID = 19134;

        public const short PLAYER = 0;
        public const short NPC = 1;
        public const short RADITATION = 2;
        public const short MINE = 3;
        public const short MISC = 4;
        public const short BATTLESTATION = 5;

        public short cause = 0;

        public DestructionTypeModule(short pCause)
        {
            this.cause = pCause;
        }

        public byte[] write()
        {
            var param1 = new ByteArray(ID);
            param1.writeShort(26620);
            param1.writeShort(this.cause);
            return param1.Message.ToArray();
        }
    }
}

File: DarkOrbit 9.0/DarkOrbit 9.0 Emulator/src/simulator/netty/serverCommands/DestructionTypeModule.java
package simulator.netty.serverCommands;

import java.io.DataOutputStream;
import java.io.IOException;

/**
 Created by Shock & Pedro on 30-03-2015.
 */
public class DestructionTypeModule {

    public static short PLAYER        = 0;
    public static short NPC           = 1;
    public static short RADITATION    = 2;
    public static short MINE          = 3;
    public static short MISC          = 4;
    public static short BATTLESTATION = 5;

    public static int ID = 8935;

    public short mCause = 0;

    public DestructionTypeModule(short pCause) {
        this.mCause = pCause;
    }

    public void write(DataOutputStream param1) {
        try {
            param1.writeShort(ID);
            this.writeInternal(param1);
        } catch (IOException e) {
        }
    }

    protected void writeInternal(DataOutputStream param1) {
        try {
            param1.writeShort(-20405);
            param1.writeShort(this.mCause);
        } catch (IOException e) {
        }
    }
}
